### PR TITLE
F#17 grasp stability measurment package

### PR DESCRIPTION
**Pull request by JamesAndrews3**
There should be no conflicts as this is all new content within it's own sub-directory.

### CODE REVIEW

File: sr_grasp_stability/src/superseded/tf_computation.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        () -> return (dict, dict)
+        """
+        # finger-tip frames:
+        # '/rh_rftip', '/rh_lftip', '/rh_mftip', '/rh_thtip', '/rh_fftip',
+
+        exc = False
+        # Get transforms from forearm to distal #
+        try:
+            (self.trans['rh_ffdistal'], self.rot['rh_ffdistal']) = self.tf_listener.lookupTransform(
```
Comment (toliver) [inline sr_grasp_stability/src/superseded/tf_computation.py:32]: I would like the fingertip names to be configurable.

The reason is that we need all this to be compatible with hand H as well.

**Comment by toliver** [comment]
sr_grasp_stability/__init__.py shouldn't be necessary

**Comment by toliver** [comment]
sr_grasp_stability/script/__init__.py
Shouldn't be necessary

**Comment by toliver** [comment]
sr_grasp_stability/src/__init__.py shouldn't be necessary

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import numpy as np
+
+# class Polygon:
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: Can we remove all commented out code that doesn't provide useful information?

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+def poly_3d_area(finger_tips):
+    """
+    Polygon area in 3D for Hand E (Five fingers)
+    """
+    polygon_area = 0.0
+
+    # 'rh_fftip', 'rh_mftip', 'rh_rftip',    'rh_lftip', 'rh_thtip'
+
+    xa, ya, za = finger_tips['rh_fftip']  # 1st finger-tip
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: Please see my other comment. Let's not assume fixed names for these links (we want this to be compatible with hand H)

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    xb, yb, zb = finger_tips['rh_mftip']  # 2nd finger-tip
+    xc, yc, zc = finger_tips['rh_rftip']  # 3rd finger-tip
+
+    a = np.array([xa, ya, za])
+    b = np.array([xb, yb, zb])
+    c = np.array([xc, yc, zc])
+
+    polygon_area += find_triangle_area(a, b, c)
+
+    xb, yb, zb = finger_tips['rh_lftip']  # 4nd finger-tip
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: And do not assume 5 fingers. Hand H has only 3.

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    xc, yc, zc = finger_tips['rh_thtip']  # 5nd finger-tip
+    c = np.array([xc, yc, zc])
+
+    polygon_area += find_triangle_area(a, b, c)
+
+    return polygon_area
+
+""" SUPERSEDED BY poly_3d_area
+Computes the polygon area of the grasp
+"""
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: Remove commented out code. Put it in a separate branch if you will use it later.

File: sr_grasp_stability/src/sr_grasp_stability/tf2_computation.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        # self. rot = {}
+        self.tfBuffer = tf2_ros.Buffer()
+        self.listener = tf2_ros.TransformListener(self.tfBuffer)
+
+    def get_finger_tips(self, ref_frame):
+        """
+        Get fingertip transformations
+
+        () -> return (dict, dict)
+        """
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/tf2_computation.py]: For any function that is documented please use the standard format. See example here 

https://github.com/shadow-robot/sr_interface/blob/indigo-devel/sr_robot_commander/src/sr_robot_commander/sr_robot_commander.py#L240

**Review by toliver — CHANGES_REQUESTED** [review]

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    for keys, values in finger_tips.items():
+        x, y, z = values
+        sum_x += x
+        sum_y += y
+        sum_z += z
+
+    return (sum_x/n), (sum_y/n), (sum_z/n)
+
+
+def find_cosine_angle(a, b, c):
```
Comment (toliver) [inline sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: This function is not used

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py (hunk 2 above)
Comment (JamesAndrews3) [reply sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: Should this be a class or just have methods? The difference is you have to instantiate an object, which isn't necessary in this case.

Comment (JamesAndrews3) [reply sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: @azompas thought a class would be better, like the rest...

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py (hunk 5 above)
Comment (JamesAndrews3) [reply sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: Makes sense

File: sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py (hunk 7 above)
Comment (JamesAndrews3) [reply sr_grasp_stability/src/sr_grasp_stability/polygon_computation.py]: I will create a new branch which includes work in progress, including this, and remove it for this branch to be merged.

**Pull request closed without merging**